lucene/analysis/common/src/java/org/apache/lucene/analysis/hunspell/Dictionary.java

```diff
@@ -1240,6 +1243,19 @@ void appendFlag(char flag, StringBuilder to) {
     }
   }
 
+  /** Used for FLAG UTF-8 on non-UTF-8 encoding */
```

Comment:
This is quite unfortunate indeed. Do I understand this right that the dictionary data is encoded as iso8859-1 but flags need to be interpreted as UTF-8? I would rename the class as Iso8859_1AsUtf8Strategy or at least provide some explanation on what's happening here in the javadoc (even if it's private).

Reply:
That's my current understanding, yes. I'll add javadoc.